File: typescript/004-resistor-color-duo/resistor-color-duo.ts
export const COLORS: string[] = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'] as const

export function decodedValue([tensColor, unitsColor]: string[]): number {
  return (COLORS.indexOf(tensColor) * 10) + COLORS.indexOf(unitsColor)
}

File: python/035-resistor-color-duo/resistor_color_duo.py
COLOR_CODE = {
    'black': 0,
    'brown': 1,
    'red': 2,
    'orange': 3,
    'yellow': 4,
    'green': 5,
    'blue': 6,
    'violet': 7,
    'grey': 8,
    'white': 9
}

def value(colors):
    if not isinstance(colors, list):
        raise TypeError('Enter the colors in an array')
    if len(colors) <= 1:
        raise ValueError('Enter two colors')
    if (colors[0] not in COLOR_CODE) or (colors[1] not in COLOR_CODE):
        raise ValueError('At least one of the colors entered isn\'t in the code')
    return COLOR_CODE[colors[0]] * 10 + COLOR_CODE[colors[1]]
